### ISSUE TITLE
String interpolation does not work

### ISSUE THREAD

**Issue by vitalyster** [vlang/v#741]
**V version: 0.1.7** 
**OS: Linux** 
**C compiler version: gcc 6.3.0** 

**What did you do?**
```
name := 'John'

println('Hello, $name')
```

**What did you expect to see?**
`Hello, John`
 
**What did you see instead?**
```
name
println

0
pass=1 fn=``
panic: hello.v:3
cannot interpolate this value
```

**Comment by AtjonTV** [comment]
I just checked and this issue only comes up when the above code is written without a wrapping main.

This works just fine:

```go
fn main() {
    name := 'John'
    println('Hello, $name')
}
```

```
$ v run t.v 
  ============running t==============================
  Hello, John
```

**Comment by thisisshub** [comment]
if you wanna do it without `fn main ()` then just 
use string datatype like this
```
name := 'John'
println('Hello '+ string(name))
```

**Comment by vitalyster** [comment]
I want to have string interpolation without main :)

**Comment by vitalyster** [comment]
Looks fixed in 0.1.8

**Comment by medvednikov** [comment]
Yes it was fixed in 0.1.8 :)

Finally

**Comment by madranet** [comment]
I'm seeing this again with  `V 0.1.25 a2838ae`

```V
import os

//fn main()
//{
// some vars
homepath := os.getenv("HOME")
configpath := "$homepath/.ssb/scutty.cfg"
println(configpath)
//}
```
```
v run tester.v
tester.v:7:41: cannot interpolate this value
    5| // some vars
    6| homepath := os.getenv("HOME")
    7| configpath := "$homepath/.ssb/scutty.cfg"
                                               ^
    8| println(configpath)
    9| //}
```
Wrap it in a main function and it works again:

```V
import os

fn main()
{
// some vars
homepath := os.getenv("HOME")
configpath := "$homepath/.ssb/scutty.cfg"
println(configpath)
}
```

```
v run tester.v
/Users/madra/.ssb/scutty.cfg
```
